### ISSUE TITLE
[Feature Request] Secret Chats

### ISSUE THREAD

**Issue by ayekyk** [kotatogram/kotatogram-desktop#8]
### Is your feature request related to a problem?
No

### Describe the solution you'd like
Add ability to use and manage secret chats

**Comment by RadRussianRus** [comment]
It would be nice to see in TDesktop, actually.

The thing is that there is no local message database for now. It may be implemented in future, but I can't say when.

**Comment by ilya-fedin** [comment]
> The thing is that there is no local message database for now.

I don't think that secret chats is something we can do. Even tdesktop developer can't implement it because it requires rewrite a big part of tdesktop. Secret chats need a lot of effort to support, it's not just message database. It's so hard so that even tdesktop developer says it's easier to rewrite the app in tdlib than just to implement them. I don't think we have enough knowledge or power to implement that in any way.

**Comment by dm17** [comment]
https://github.com/marcovelon/tdesktop/issues/2 - it is happening and there are resources being raised

**Comment by baptx** [comment]
I found a Telegram Desktop issue https://github.com/telegramdesktop/tdesktop/issues/871 (mentioned on https://github.com/telegramdesktop/tdesktop/issues/17216).
Someone said that Cutegram supports encrypted chats but it does not seem maintained anymore. I also found a Telegram desktop fork that claims to have added encrypted chats but it is not popular so the source code should be carefully reviewed:
https://github.com/Telegram-Encrypted/tdesktop-encrypted
Maybe the work done on these alternative clients can be integrated in kotatogram.
Another solution I was thinking about is to use Telegram web version with a web browser addon that encrypts and decrypts messages with PGP, like Mailvelope is doing. But it will only work if other users we are chatting with have the same setup.

**Comment by ilya-fedin** [comment]
You could notice kotatogram is almost dead due to too much features to rebase already :(

**Comment by dm17** [comment]
In my view, if someone made a high quality FOSS implementation of this and upgraded the clients to have some seamless "federated" functionality (still connect to main server, but can add/invite to other servers), then I think secret chats would end up getting implemented sooner and get used/tested/maintained by more people: https://github.com/loyldg/mytelegram
